src/main/java/com/semmle/jira/addon/workflow/ResolutionValidatorFactory.java

```diff
@@ -0,0 +1,82 @@
+package com.semmle.jira.addon.workflow;
+
+import com.atlassian.jira.component.ComponentAccessor;
+import com.atlassian.jira.config.ResolutionManager;
+import com.atlassian.jira.issue.resolution.Resolution;
+import com.atlassian.jira.plugin.workflow.AbstractWorkflowPluginFactory;
+import com.atlassian.jira.plugin.workflow.WorkflowPluginConditionFactory;
+import com.atlassian.jira.plugin.workflow.WorkflowPluginValidatorFactory;
+import com.opensymphony.workflow.loader.AbstractDescriptor;
+import com.opensymphony.workflow.loader.ConditionDescriptor;
+import com.opensymphony.workflow.loader.ValidatorDescriptor;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@SuppressWarnings({"unchecked", "rawtypes"})
+public class ResolutionValidatorFactory extends AbstractWorkflowPluginFactory
+    implements WorkflowPluginValidatorFactory, WorkflowPluginConditionFactory {
+
+  private static final String FIELD_RESOLUTION_NAME = "resolutionName";
+  private static final String FIELD_OPERATOR_NAME = "operatorName";
+  private static final String FIELD_RESOLUTIONS = "resolutions";
+  private static final String FIELD_OPERATORS = "operators";
+  private Map<String, String> resolutions = new LinkedHashMap<>();
+
+  public ResolutionValidatorFactory() {
+    resolutions =
+        ComponentAccessor.getComponent(ResolutionManager.class).getResolutions().stream()
+            .collect(Collectors.toMap(Resolution::getId, Resolution::getName));
+  }
+
+  @Override
+  protected void getVelocityParamsForInput(Map velocityParams) {
+    velocityParams.put(FIELD_RESOLUTIONS, resolutions);
+    velocityParams.put(FIELD_OPERATORS, Operator.values());
+  }
+
+  @Override
+  protected void getVelocityParamsForEdit(Map velocityParams, AbstractDescriptor descriptor) {
+    getVelocityParamsForInput(velocityParams);
+    getVelocityParamsForView(velocityParams, descriptor);
+  }
+
+  @Override
+  protected void getVelocityParamsForView(Map velocityParams, AbstractDescriptor descriptor) {
+    Map<String, String> arguments;
+    if (descriptor instanceof ValidatorDescriptor) {
+      arguments = ((ValidatorDescriptor) descriptor).getArgs();
+    } else if (descriptor instanceof ConditionDescriptor) {
+      arguments = ((ConditionDescriptor) descriptor).getArgs();
+    } else {
+      throw new IllegalArgumentException(
+          "Descriptor must be a ConditionDescriptor or ValidatorDescriptor.");
+    }
+    String resolutionId = arguments.get(ResolutionValidator.FIELD_RESOLUTION);
```

Comment:
This should be `CheckResolutionFunction.FIELD_RESOLUTION` The same with the other instances of this